src/components/BlockSelector/BlockSelector.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <ToggleInput
+    :id="id"
+    class="block-selector"
+    :class="[
+      {
+        'block-selector--selected': checked,
+        'block-selector--focus': isFocused,
+        'block-selector--disabled': disabled,
+      },
+      `block-selector--${labelPosition}`,
+    ]"
+    :type="type"
+    :data-checked="`${checked}`"
+    :checked="checked"
+    :disabled="disabled"
+    :label-position="labelPosition"
+    :name="name"
+    @focus="isFocused = true"
+    @blur="isFocused = false"
+    @input="$emit('input', $event.target.checked)"
+  >
+    <div class="block-selector__text">
+      <div v-if="!!$slots.label || label">
+        <TypeBody
+          tag="div"
+          variant="medium"
+          weight="strong"
+          class="block-selector__text__label"
+        >
+          <slot name="label">
+            {{ label }}
+          </slot>
+        </TypeBody>
+      </div>
```

Comment:
Within Pollen internally, there's actually font classes that can be used instead of the `TypeX` components. So this can be simplified:

```suggestion
      <div v-if="!!$slots.label || label" class="block-selector__text__label">
        <slot name="label">
          {{ label }}
        </slot>
      </div>
```

and in the styles:

```
.block-selector__text__label {
  @apply font-body-medium font-semibold text-gray-0;
}
```